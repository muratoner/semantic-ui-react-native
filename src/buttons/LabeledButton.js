import PropTypes from 'prop-types';
import React from 'react';
import {
	ActivityIndicator,
	StyleSheet,
	Text,
	TouchableOpacity,
	View
} from 'react-native';
import { LabeledButtonProps } from '../';
import Icon from '../icons/Icon';
import ButtonColor from './ButtonColor';

const LabeledButton = ({
	circular,
	color,
	disabled,
	fluid,
	indicatorStyle,
	label,
	labelIcon,
	labelIconType,
	labelRight,
	loading,
	pointing,
	onPress,
	style,
	title,
	type
}: LabeledButtonProps) => {
	const bgColor = ButtonColor.getBgColor(color);
	const textColor = ButtonColor.getTextColor(color);
	const outline = type == 'outline';

	const getValueColor = () => {
		switch (color) {
		case 'red':
			return '#db2828';
		case 'primary':
			return '#2185d0';
		default:
			return 'rgba(0,0,0,.6)';
		}
	};

	const getLabel = () => {
		const iconStyle = {
			marginRight: label ? 5 : 0
		};
		return (
			<View
				style={StyleSheet.flatten([
					styles.labelContainer,
					{
						backgroundColor: outline ? '#fff' : bgColor,
						borderColor: outline ? bgColor : '#fff',
						borderRightWidth: outline ? 1 : 0
					}
				])}
			>
				{labelIcon && labelIconType && (
					<Icon
						name={labelIcon}
						type={labelIconType}
						color={outline ? bgColor : textColor}
						style={StyleSheet.flatten([styles.icon, iconStyle])}
					/>
				)}
				{label && (
					<Text
						style={StyleSheet.flatten([
							styles.label,
							{
								color: outline ? bgColor : textColor
							}
						])}
					>
						{label}
					</Text>
				)}
				{labelRight && getPointing()}
			</View>
		);
	};

	const getPointing = () => {
		const customStyle = {
			borderColor: outline ? bgColor : '#fff',
			transform: [
				{ rotate: '45deg' },
				{ translateX: labelRight ? -6 : -3 },
				{ translateY: labelRight ? 0 : 4 }
			]
		};
		return pointing ? (
			<View style={StyleSheet.flatten([styles.pointing, customStyle])} />
		) : null;
	};

	return (
		<TouchableOpacity
			onPress={disabled || loading ? null : onPress}
			disabled={disabled}
			style={StyleSheet.flatten([
				styles.button,
				{
					borderColor: bgColor,
					borderRadius: circular ? 50 : 3,
					width: fluid ? '100%' : undefined,
					opacity: disabled ? 0.45 : 1
				},
				style
			])}
		>
			{!labelRight && getLabel()}
			{title && !loading && (
				<Text
					style={StyleSheet.flatten([
						styles.title,
						{
							color: getValueColor()
						}
					])}
				>
					{!labelRight && getPointing()}
					{title}
				</Text>
			)}
			{loading && (
				<ActivityIndicator
					style={StyleSheet.flatten([styles.indicator, indicatorStyle])}
				/>
			)}
			{labelRight && getLabel()}
		</TouchableOpacity>
	);
};

const styles = StyleSheet.create({
	labelContainer: {
		height: '100%',
		paddingHorizontal: 21,
		flexDirection: 'row',
		alignItems: 'center'
	},
	label: {
		fontWeight: 'bold'
	},
	button: {
		alignItems: 'center',
		justifyContent: 'center',
		flexDirection: 'row',
		overflow: 'hidden',
		borderWidth: 1
	},
	icon: {
		fontSize: 15
	},
	indicator: {
		paddingHorizontal: 10,
		paddingVertical: 10,
		flex: 1
	},
	title: {
		fontWeight: 'bold',
		paddingHorizontal: 10,
		paddingVertical: 10,
		backgroundColor: 'white',
		flex: 1
	},
	pointing: {
		position: 'absolute',
		width: 10,
		height: 10,
		backgroundColor: '#fff',
		borderBottomWidth: 1,
		borderLeftWidth: 1,
		top: '50%'
	}
});

LabeledButton.propTypes = {
	title: PropTypes.string,
	color: PropTypes.oneOf([
		'primary',
		'secondary',
		'red',
		'positive',
		'negative'
	]),
	iconName: PropTypes.string,
	iconType: PropTypes.oneOf([
		'AntDesign',
		'Entypo',
		'EvilIcons',
		'Feather',
		'FontAwesome',
		'FontAwesome5',
		'Fontisto',
		'Foundation',
		'Ionicons',
		'MaterialCommunityIcons',
		'MaterialIcons',
		'Octicons',
		'SimpleLineIcons',
		'Zocial'
	])
};

export default LabeledButton;
