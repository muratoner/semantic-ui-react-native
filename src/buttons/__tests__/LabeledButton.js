import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import LabeledButton from '../LabeledButton';

describe('Labeled Button Component', () => {
	it('should render without issues', () => {
		const component = shallow(<LabeledButton />);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with title prop', () => {
		const component = shallow(<LabeledButton title="Save" />);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should not be call onPress events when disabled is true', () => {
		const onPress = jest.fn();
		const component = shallow(<LabeledButton onPress={onPress} disabled />);

		expect(component.get(0).props.style).toHaveProperty('opacity', 0.45);

		// Simulate press action
		component.simulate('press');

		expect(onPress).not.toHaveBeenCalled();
	});

	it('should not be call onPress events when loading is true', () => {
		const onPress = jest.fn();
		const component = shallow(<LabeledButton onPress={onPress} loading />);

		expect(component.find('ActivityIndicator')).toHaveLength(1);

		// Simulate press action
		component.simulate('press');

		expect(onPress).not.toHaveBeenCalled();
	});

	it('should be call onPress events', () => {
		const onPress = jest.fn();
		const component = shallow(<LabeledButton onPress={onPress} />);

		// Simulate press action
		component.simulate('press');

		expect(onPress).toHaveBeenCalled();
	});

	it('should render with title and circular props', () => {
		const component = shallow(<LabeledButton title="Save" circular />);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with title, circular and disabled props', () => {
		const component = shallow(<LabeledButton title="Save" circular disabled />);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with title, circular, disabled and fluid props', () => {
		const component = shallow(
			<LabeledButton title="Save" circular disabled fluid />
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with title, circular, disabled, fluid and outline props', () => {
		const component = shallow(
			<LabeledButton title="Save" circular disabled fluid outline />
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with title, circular, disabled, fluid, outline style props', () => {
		const component = shallow(
			<LabeledButton
				title="Save"
				circular
				disabled
				fluid
				outline
				style={{
					height: 50
				}}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with title, circular, disabled, fluid, outline and style props', () => {
		const component = shallow(
			<LabeledButton
				title="Save"
				circular
				disabled
				fluid
				outline
				style={{
					height: 50
				}}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with title, circular, disabled, fluid, outline, style and color props', () => {
		const component = shallow(
			<LabeledButton
				title="Save"
				circular
				disabled
				fluid
				outline
				style={{
					height: 50
				}}
				color="red"
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with title, circular, disabled, fluid, outline, style and color props', () => {
		const component = shallow(
			<LabeledButton
				title="Save"
				circular
				disabled
				fluid
				outline
				style={{
					height: 50
				}}
				color="red"
				iconName="user"
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with title, circular, disabled, fluid, outline, style, color and iconName props', () => {
		const component = shallow(
			<LabeledButton
				title="Save"
				circular
				disabled
				fluid
				outline
				style={{
					height: 50
				}}
				color="red"
				iconName="user"
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with circular, disabled, fluid, outline, style, color, iconName and iconType props', () => {
		const component = shallow(
			<LabeledButton
				circular
				disabled
				fluid
				outline
				style={{
					height: 50
				}}
				color="red"
				iconName="user"
				iconType="Ionicons"
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with circular, disabled, fluid, outline, style, color, iconName, iconType and pointing props', () => {
		const component = shallow(
			<LabeledButton
				circular
				disabled
				fluid
				outline
				style={{
					height: 50
				}}
				color="red"
				iconName="user"
				iconType="Ionicons"
				pointing
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with circular, disabled, fluid, outline, style, color, iconName, iconType and pointing props', () => {
		const component = shallow(
			<LabeledButton
				loading
				circular
				disabled
				fluid
				outline
				style={{
					height: 50
				}}
				color="red"
				iconName="user"
				iconType="Ionicons"
				pointing
				label="Forks"
				labelIcon="fork"
				labelIconType="Ionicons"
				labelRight
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with color, title, label, labelIcon and labelIconType props', () => {
		const component = shallow(
			<LabeledButton
				color="primary"
				title="1,048"
				label="Forks"
				labelIcon="fork"
				labelIconType="Ionicons"
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});
});
