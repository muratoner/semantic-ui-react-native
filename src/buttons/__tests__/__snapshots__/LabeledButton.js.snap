// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labeled Button Component should render with circular, disabled, fluid, outline, style, color, iconName and iconType props 1`] = `
<ForwardRef
  disabled={true}
  onPress={null}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#db2828",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 0.45,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#db2828",
        "borderRightWidth": 1,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  />
</ForwardRef>
`;

exports[`Labeled Button Component should render with circular, disabled, fluid, outline, style, color, iconName, iconType and pointing props 1`] = `
<ForwardRef
  disabled={true}
  onPress={null}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#db2828",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 0.45,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#db2828",
        "borderRightWidth": 1,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  />
</ForwardRef>
`;

exports[`Labeled Button Component should render with circular, disabled, fluid, outline, style, color, iconName, iconType and pointing props 2`] = `
<ForwardRef
  disabled={true}
  onPress={null}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#db2828",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 0.45,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#db2828",
        "borderRightWidth": 1,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  >
    <Icon
      color="#db2828"
      name="fork"
      style={
        Object {
          "fontSize": 15,
          "marginRight": 5,
        }
      }
      type="Ionicons"
    />
    <Text
      style={
        Object {
          "color": "#db2828",
          "fontWeight": "bold",
        }
      }
    >
      Forks
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#db2828",
          "borderLeftWidth": 1,
          "height": 10,
          "position": "absolute",
          "top": "50%",
          "transform": Array [
            Object {
              "rotate": "45deg",
            },
            Object {
              "translateX": -6,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": 10,
        }
      }
    />
  </View>
</ForwardRef>
`;

exports[`Labeled Button Component should render with color, title, label, labelIcon and labelIconType props 1`] = `
<ForwardRef
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#2185d0",
      "borderRadius": 3,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "overflow": "hidden",
      "width": undefined,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2185d0",
        "borderColor": "#fff",
        "borderRightWidth": 0,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  >
    <Icon
      color="#fff"
      name="fork"
      style={
        Object {
          "fontSize": 15,
          "marginRight": 5,
        }
      }
      type="Ionicons"
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "bold",
        }
      }
    >
      Forks
    </Text>
  </View>
  <Text
    style={
      Object {
        "backgroundColor": "white",
        "color": "rgba(0,0,0,.6)",
        "flex": 1,
        "fontWeight": "bold",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    1,048
  </Text>
</ForwardRef>
`;

exports[`Labeled Button Component should render with title and circular props 1`] = `
<ForwardRef
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#e0e1e2",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "overflow": "hidden",
      "width": undefined,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e0e1e2",
        "borderColor": "#fff",
        "borderRightWidth": 0,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "white",
        "color": "rgba(0,0,0,.6)",
        "flex": 1,
        "fontWeight": "bold",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    Save
  </Text>
</ForwardRef>
`;

exports[`Labeled Button Component should render with title prop 1`] = `
<ForwardRef
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#e0e1e2",
      "borderRadius": 3,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "overflow": "hidden",
      "width": undefined,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e0e1e2",
        "borderColor": "#fff",
        "borderRightWidth": 0,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "white",
        "color": "rgba(0,0,0,.6)",
        "flex": 1,
        "fontWeight": "bold",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    Save
  </Text>
</ForwardRef>
`;

exports[`Labeled Button Component should render with title, circular and disabled props 1`] = `
<ForwardRef
  disabled={true}
  onPress={null}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#e0e1e2",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 0.45,
      "overflow": "hidden",
      "width": undefined,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e0e1e2",
        "borderColor": "#fff",
        "borderRightWidth": 0,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "white",
        "color": "rgba(0,0,0,.6)",
        "flex": 1,
        "fontWeight": "bold",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    Save
  </Text>
</ForwardRef>
`;

exports[`Labeled Button Component should render with title, circular, disabled and fluid props 1`] = `
<ForwardRef
  disabled={true}
  onPress={null}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#e0e1e2",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 0.45,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e0e1e2",
        "borderColor": "#fff",
        "borderRightWidth": 0,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "white",
        "color": "rgba(0,0,0,.6)",
        "flex": 1,
        "fontWeight": "bold",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    Save
  </Text>
</ForwardRef>
`;

exports[`Labeled Button Component should render with title, circular, disabled, fluid and outline props 1`] = `
<ForwardRef
  disabled={true}
  onPress={null}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#e0e1e2",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 0.45,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#e0e1e2",
        "borderRightWidth": 1,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "white",
        "color": "rgba(0,0,0,.6)",
        "flex": 1,
        "fontWeight": "bold",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    Save
  </Text>
</ForwardRef>
`;

exports[`Labeled Button Component should render with title, circular, disabled, fluid, outline and style props 1`] = `
<ForwardRef
  disabled={true}
  onPress={null}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#e0e1e2",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 0.45,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#e0e1e2",
        "borderRightWidth": 1,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "white",
        "color": "rgba(0,0,0,.6)",
        "flex": 1,
        "fontWeight": "bold",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    Save
  </Text>
</ForwardRef>
`;

exports[`Labeled Button Component should render with title, circular, disabled, fluid, outline style props 1`] = `
<ForwardRef
  disabled={true}
  onPress={null}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#e0e1e2",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 0.45,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#e0e1e2",
        "borderRightWidth": 1,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "white",
        "color": "rgba(0,0,0,.6)",
        "flex": 1,
        "fontWeight": "bold",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    Save
  </Text>
</ForwardRef>
`;

exports[`Labeled Button Component should render with title, circular, disabled, fluid, outline, style and color props 1`] = `
<ForwardRef
  disabled={true}
  onPress={null}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#db2828",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 0.45,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#db2828",
        "borderRightWidth": 1,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "white",
        "color": "rgba(0,0,0,.6)",
        "flex": 1,
        "fontWeight": "bold",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    Save
  </Text>
</ForwardRef>
`;

exports[`Labeled Button Component should render with title, circular, disabled, fluid, outline, style and color props 2`] = `
<ForwardRef
  disabled={true}
  onPress={null}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#db2828",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 0.45,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#db2828",
        "borderRightWidth": 1,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "white",
        "color": "rgba(0,0,0,.6)",
        "flex": 1,
        "fontWeight": "bold",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    Save
  </Text>
</ForwardRef>
`;

exports[`Labeled Button Component should render with title, circular, disabled, fluid, outline, style, color and iconName props 1`] = `
<ForwardRef
  disabled={true}
  onPress={null}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#db2828",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 0.45,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#db2828",
        "borderRightWidth": 1,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "white",
        "color": "rgba(0,0,0,.6)",
        "flex": 1,
        "fontWeight": "bold",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    Save
  </Text>
</ForwardRef>
`;

exports[`Labeled Button Component should render without issues 1`] = `
<ForwardRef
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#e0e1e2",
      "borderRadius": 3,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "overflow": "hidden",
      "width": undefined,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e0e1e2",
        "borderColor": "#fff",
        "borderRightWidth": 0,
        "flexDirection": "row",
        "height": "100%",
        "paddingHorizontal": 21,
      }
    }
  />
</ForwardRef>
`;
